### PR TITLE
Delete networks only when gone from device with cleanup

### PR DESCRIPTION
**Pull request by sebageek**

### CODE REVIEW

**Comment by joker-at-work** [comment]
commit "Only deprovision vlan if no port…"
> are still in used we
"are still in use, we"

File: networking_arista/ml2/mechanism_arista.py [hunk 1]
```diff
@@ -859,7 +859,17 @@ def _delete_port(self, context, port, host, tenant_id, segments=None):
                     )
 
         try:
-            if not cfg.CONF.ml2_arista.skip_unplug:
+            device_ports = db_lib.get_bm_ports_for_device(
+                context._plugin_context, device_id)
+            port_net_in_use = False
+            for device_port in device_ports:
+                if device_port.id != port_id and \
+                        device_port.network_id == network_id:
```
Comment (joker-at-work) [inline networking_arista/ml2/mechanism_arista.py:867]: I'd write this as
```python
if device_port.id = port_id:
  continue
if device_port.network_id != network_id:
  continue
```
But probably taste ;)

File: networking_arista/ml2/arista_sec_gp.py [hunk 2]
```diff
@@ -325,6 +329,56 @@ def _get_interface_membership(self, server, ports):
...
+        # find mlag neighbor
+        mlag_detail = server(["show mlag detail"])
+        if not (mlag_detail and mlag_detail[0]):
+            LOG.warning("Could not fetch mlag details on %s", server)
+            return None
+        neigh = EUI(mlag_detail[0]['detail']['peerMacAddress'])
+        neigh_server = self._get_server(self, switch_id=neigh)
+        if not neigh_server:
+            LOG.warning("Could not find mlag neighbor %s in switch list",
+                        neigh, server)
```
Comment (joker-at-work) [inline networking_arista/ml2/arista_sec_gp.py]: Doesn't that log line need another `%s` somewhere?

**Review by joker-at-work — CHANGES_REQUESTED** [review]
One little thing, that should pretty much never trigger, but anyways.

File: networking_arista/ml2/arista_sec_gp.py (hunk 2 above)
Comment (sebageek) [reply networking_arista/ml2/arista_sec_gp.py]: Yes, nice find! Fixed it

**Review by joker-at-work — APPROVED** [review]

**Pull request merged**